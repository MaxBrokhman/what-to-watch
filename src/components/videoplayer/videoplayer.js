import React from 'react';
// eslint-disable-next-line
export const Videoplayer = ({src}) => (
  <video src={src} width="280" height="175" autoPlay muted></video>
);
