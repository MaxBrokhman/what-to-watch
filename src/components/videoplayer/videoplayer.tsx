import React, {
  forwardRef,
  VideoHTMLAttributes,
  MutableRefObject,
} from 'react';

export const VideoplayerComponent = (props: VideoHTMLAttributes<{}>, ref: MutableRefObject<HTMLVideoElement>): JSX.Element => (
  <video {...props} ref={ref}></video>
);

export const Videoplayer = forwardRef(VideoplayerComponent);
