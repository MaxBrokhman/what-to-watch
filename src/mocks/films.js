/* eslint-disable */
export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `fantasy`,
    background_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    run_time: 88,
    released: 2003,
    id: 1,
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/bohemian-rhapsody.jpg`,
    poster_image: `img/bohemian-rhapsody.jpg`,
    run_time: 88,
    released: 2003,
    id: 2,
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/macbeth.jpg`,
    poster_image: `img/macbeth.jpg`,
    run_time: 88,
    released: 2003,
    id: 3,
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `thriller`,
    background_image: `img/aviator.jpg`,
    poster_image: `img/aviator.jpg`,
    run_time: 88,
    released: 2003,
    id: 4,
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `comedy`,
    background_image: `img/we-need-to-talk-about-kevin.jpg`,
    poster_image: `img/we-need-to-talk-about-kevin.jpg`,
    run_time: 88,
    released: 2003,
    id: 5
  }
];
