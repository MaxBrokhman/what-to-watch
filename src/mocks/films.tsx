/* eslint-disable */
import {TFilm} from "../reducer/types";

export const films: Array<TFilm> = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `fantasy`,
    background_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    run_time: 88,
    released: 2003,
    id: 1,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/bohemian-rhapsody.jpg`,
    poster_image: `img/bohemian-rhapsody.jpg`,
    run_time: 88,
    released: 2003,
    id: 2,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `drama`,
    background_image: `img/macbeth.jpg`,
    poster_image: `img/macbeth.jpg`,
    run_time: 88,
    released: 2003,
    id: 3,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `thriller`,
    background_image: `img/aviator.jpg`,
    poster_image: `img/aviator.jpg`,
    run_time: 88,
    released: 2003,
    id: 4,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `comedy`,
    background_image: `img/we-need-to-talk-about-kevin.jpg`,
    poster_image: `img/we-need-to-talk-about-kevin.jpg`,
    run_time: 88,
    released: 2003,
    id: 5,
    director: ``,
    starring: [],
    rating: 9,
    description: ``,
  }
];
