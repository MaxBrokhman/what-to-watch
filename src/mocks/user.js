export const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver`,
  // eslint-disable-next-line
  avatar_url: `#`,
};
