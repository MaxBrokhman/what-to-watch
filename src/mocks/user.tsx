import {TUser} from "../reducer/types";

export const user: TUser = {
  email: `Oliver.conner@gmail.com`,
  name: `Oliver`,
  // eslint-disable-next-line
  avatar_url: `#`,
};
