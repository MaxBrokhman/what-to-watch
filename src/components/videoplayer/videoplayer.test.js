import React from 'react';
import renderer from 'react-test-renderer';

import {Videoplayer} from './videoplayer';
import {films} from '../../mocks/films';

it(`Renders Videoplayer component correctly`, () => {
  const tree = renderer
                .create(
                    <Videoplayer
                      src={films[0].preview}
                    />
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
