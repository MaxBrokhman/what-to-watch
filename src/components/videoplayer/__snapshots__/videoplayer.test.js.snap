// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Videoplayer component correctly 1`] = `<video />`;
